kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -406,4 +411,124 @@ class InCommitTimestampSuite extends DeltaTableWriteSuiteBase {
       Option.empty
     }
   }
+
+  test("Conflict resolution of timestamps") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = TableImpl.forPath(engine, tablePath, () => System.currentTimeMillis)
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val startTime = System.currentTimeMillis()
+      val clock = new ManualClock(startTime)
+      val txn1 = createTxn(
+        engine,
+        tablePath,
+        schema = testSchema,
+        partCols = Seq.empty,
+        clock = clock
+      )
+      clock.setTime(startTime)
+      appendData(
+        engine,
+        tablePath,
+        data = Seq(Map.empty[String, Literal] -> dataBatches2),
+        clock = clock
+      )
+      clock.setTime(startTime - 1000)
+      commitAppendData(engine, txn1, Seq(Map.empty[String, Literal] -> dataBatches1))
+      assert(
+        getInCommitTimestamp(
+          engine, table, 2).get == getInCommitTimestamp(engine, table, 1).get + 1)
```

Comment:
```suggestion
      assert(
        getInCommitTimestamp(engine, table, 2).get ===
        getInCommitTimestamp(engine, table, 1).get + 1)
```

Reply:
nit: also use named literals `version = 2` (or `2 /* version*/` for Java). Easy to read the code.